tests/factories.py

```diff
@@ -214,7 +214,10 @@ def create_repo(workspace, files=None, temporary=True):
 
     commit = response.stdout.strip()
     update_manifest(workspace)
-    workspace.manifest["repo"] = str(repo_dir)
+    if not workspace.manifest["outputs"]:
```

Comment:
Hmm, awkward workarounds like this do make me think that a root repo in the manifest may actually simplify a lot.

There's a history of what the root `repo` key used to be used for (github based releases), and I was hoping to not reuse it, as it's potentially brittle state. But I think the complexity it adds may be more costly than unlikely edge cases.

Reply:
Can a job run and create a manifest with no outputs? This is a slightly awkward workaround, but it's only for tests and made-up local manifests, to make them have at least one output. If real manifests will always have at least one output to look up, i don't think we need to worry about it. Plus, we should only ever be fetching the repo (in prod, that is) for either a workspace file or a request file, which should have a manifest entry.

Reply:
I'm going to leave this as is for now and close the job-runner PR. We can revisit later if necessary.

Reply:
Yes, sorry, was a comment on the need for it because of other decisions, not a thing to be addressed. Sorry, should have made that clearer.